plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/explorer/BaseExplorerContentProvider.java

```diff
@@ -224,11 +225,14 @@ protected void refreshViewer(final Object object) {
 		asyncViewerRefresh(new Runnable() {
 			public void run() {
 				synchronized (viewer) {
-					if(object != null) {
+					//final TreePath[] treePaths = viewer.getExpandedTreePaths();
+					if (object != null) {
 						viewer.refresh(object);
-					}else {
+					} else {
 						viewer.refresh();
 					}
+					// This line causes an endless load on linux when a node has zero children. (JCantrill)
+					//viewer.setExpandedTreePaths(treePaths);
```

Comment:
Without it, expand state is lost. Can we check which tree paths are safe to set expanded and call this method with filtered list?